plugins/aws/ec2/ebsSnapshotLifecycle.js

```diff
@@ -0,0 +1,156 @@
+        'DLM:getLifecyclePolicy', 'STS:getCallerIdentity'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var acctRegion = helpers.defaultRegion(settings);
+        var awsOrGov = helpers.defaultPartition(settings);
+        var accountId = helpers.addSource(cache, source, ['sts', 'getCallerIdentity', acctRegion, 'data']);
+
+        async.each(regions.ec2, function(region, rcb) {
+            var describeVolumes = helpers.addSource(cache, source,
+                ['ec2', 'describeVolumes', region]);
+
+            if (!describeVolumes) return rcb();
+
+            if (describeVolumes.err || !describeVolumes.data) {
+                helpers.addResult(results, 3,
+                    `Unable to describe EBS volumes: ${helpers.addError(describeVolumes)}`, region);
+                return rcb();
+            }
+
+            if (!describeVolumes.data.length) {
+                helpers.addResult(results, 0,
+                    'No EBS volumes found', region);
+                return rcb();
+            }
+
+            var describeInstances = helpers.addSource(cache, source,
+                ['ec2', 'describeInstances', region]);
+
+            if (!describeInstances || describeInstances.err || !describeInstances.data) {
+                helpers.addResult(results, 3,
+                    `Unable to EC2 instances: ${helpers.addError(describeInstances)}`, region);
+                return rcb();
+            }
+
+            var getLifecyclePolicies = helpers.addSource(cache, source,
+                ['dlm', 'getLifecyclePolicies', region]);
+
+            if (!getLifecyclePolicies || getLifecyclePolicies.err | !getLifecyclePolicies.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query DLM lifecycle policies: ${helpers.addError(getLifecyclePolicies)}`, region);
+                return rcb();
+            }
+
+            var passingVolumes = [];
+            var passingVolumeTags = {};
+            var passingInstanceTags = {};
+            getLifecyclePolicies.data.forEach(policy => {
+                var getLifecyclePolicy = helpers.addSource(cache, source,
+                    ['dlm', 'getLifecyclePolicy', region, policy.PolicyId]);
+
+                if (!getLifecyclePolicy || getLifecyclePolicy.err || !getLifecyclePolicy.data || !getLifecyclePolicy.data.Policy) {
+                    helpers.addResult(results, 3,
+                        `Unable to query lifecycle policy: ${helpers.addError(getLifecyclePolicy)}`,
+                        region, policy.PolicyId);
+                    return;
+                }
+
+                if (getLifecyclePolicy.data.Policy.State &&
+                    getLifecyclePolicy.data.Policy.State === 'ENABLED' &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags) {
```

Comment:
you are looping through this without first checking if it has anything in it
```suggestion
                    getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags && getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags.length) {
```